### PR TITLE
15728912 only show published videos

### PR DESCRIPTION
**Pull request by alvaromart**
Added new editor action, publish video. 
Show only published videos if user is not editor or if he is not querying his own videos.
Icon published video showed on top-right of video card with circular background and in video details it is activated.
Upload video redirect to user gallery, main gallery only show published videos.
In video details, user card show video uploaded, not video published.

### CODE REVIEW

File: pages/upload/upload.controller.js [hunk 1]
```diff
@@ -42,7 +42,8 @@ function UploadController(api, session, $state, $translate, productType) {
 			self.loading = true;
 			api.upload(api.url + '/video', self.file, data, function() {
 				self.loading = false;
-				$state.go('gallery');
+				console.log('video updated, redirecting to userGallery');
```
Comment (DevStarlight) [inline pages/upload/upload.controller.js]: I'd say we don't need that console.log anymore =)

File: components/video-card/video-card.js [hunk 2]
```diff
@@ -16,6 +17,16 @@ function videoCardDirective() {
...
 		scope.verified = scope.video.verified;
+		scope.owner = scope.video.owner;
+		scope.showPublishedIcon = showPublishedIcon(scope.owner, scope.published);
+	}
+
+	function showPublishedIcon(owner, published) {
+		if (session.role === 'editor' || (owner === session.id && $state.current.name === 'userGallery')) {
+			return (published === true || published === 'true');
+		} else {
+			return false;
```
Comment (DevStarlight) [inline components/video-card/video-card.js]: we can improve this code that way: 
```
if (session.role === 'editor' || (owner === session.id && $state.current.name === 'userGallery')) {
   return (published === true || published === 'true');
}
return false;
```

**Review by DevStarlight — APPROVED** [review]

File: components/video-card/video-card.js (hunk 2 above)
Comment (alvaromart) [reply components/video-card/video-card.js]: +1, in video-detail-controller we have the same mistake.

File: pages/upload/upload.controller.js (hunk 1 above)
Comment (alvaromart) [reply pages/upload/upload.controller.js]: Thanks.

**Pull request merged**